test/sing_for_needs_web/schema/query/causes_test.exs

```diff
@@ -181,4 +181,92 @@ defmodule SingForNeeds.Schema.Query.CauseTest do
 
     assert expected_result == json_response(conn, 200)
   end
+
+  test "causes ending soon and limit passed to causes query" do
+    causes_ending_soon_with_limit_query = """
+      query($scope: String, $limit: Int) {
+          causes(scope: $scope, limit: $limit) {
+              name
+          }
+      }
+    """
+
+    twenty_days_from_now = Timex.add(Timex.now(), Timex.Duration.from_days(20))
+    thirty_days_from_now = Timex.add(Timex.now(), Timex.Duration.from_days(30))
+    fifteen_days_from_now = Timex.add(Timex.now(), Timex.Duration.from_days(15))
+    five_days_ago = Timex.add(Timex.now(), Timex.Duration.from_days(-5))
+    nine_days_ago = Timex.add(Timex.now(), Timex.Duration.from_days(-9))
+
+    insert(:cause, %{name: "Cause ends in 30 days", end_date: thirty_days_from_now})
```

Comment:
Agree, let me refactor these to a helper file...

Reply:
It's also fine to leave it and circle back for it in a dedicated refactor.

But since factories already exist in the project, I think you can justify moving those Causes used by these tests in this PR.